src/Servers/Kestrel/Core/src/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -189,7 +189,7 @@ public void WriteResponseHeaders(int streamId, int statusCode, Http2HeadersFrame
                     var done = HPackHeaderWriter.BeginEncodeHeaders(statusCode, _hpackEncoder, _headersEnumerator, buffer, out var payloadLength);
                     FinishWritingHeaders(streamId, payloadLength, done);
                 }
-                catch (HPackEncodingException hex)
+                catch (Exception hex) // Any exception from the HPack encoder can leave the dynamic table in a corrupt state.
```

Comment:
```suggestion
                catch (Exception ex) // Any exception from the HPack encoder can leave the dynamic table in a corrupt state.
```

Reply:
Maybe point out that custom encoders are a reason we cannot be more certain about what Exception types might be thrown?